### PR TITLE
[ADD] Add new module to manage change product quantity reason

### PR DESCRIPTION
**Pull request by rousseldenis**

### CODE REVIEW

File: stock_change_qty_reason/README.rst [hunk 1]
```diff
@@ -0,0 +1,47 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :alt: License: AGPL-3
+
+Stock Change Quantity Reason
```
Comment (elicoidal) [inline stock_change_qty_reason/README.rst:4]: @rousseldenis   Thanks for your contribution to the OCA. 
Let me introduce to you some basic information about contribution in OCA and I will review other points in more comments inside the code..
In general, please follow the principles: 
- Contribution guidelines: https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md
- Please follow the OCA README template:
  https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst
- We recommend to include a screenshot as well to help new comers (nice to have)
- Please use simple headers in all py files following OCA templates
  https://github.com/OCA/maintainer-tools/blob/master/template/module/__openerp__.py#L3

In here, please add a line before the document title

File: stock_change_qty_reason/README.rst [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+============================
+
+This module was written to extend the product stock management. It allows to set a reason
```
Comment (elicoidal) [inline stock_change_qty_reason/README.rst]: This module extends the product stock management and allows to set a reason
in the wizard when changing the product quantity.

File: stock_change_qty_reason/README.rst [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+on the wizard changing product quantity
+
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues <https://github.com/OCA/purchase-workflow/issues>`_.
+In case of trouble, please check there if your issue has already been reported.
+If you spotted it first, help us smashing it by providing a detailed and welcomed feedback
```
Comment (elicoidal) [inline stock_change_qty_reason/README.rst:15]: You can simplify following last README template

File: stock_change_qty_reason/README.rst [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+   :alt: Odoo Community Association
+   :target: https://odoo-community.org
+
+This module is maintained by the OCA.
+
+OCA, or the Odoo Community Association, is a nonprofit organization whose
+mission is to support the collaborative development of Odoo features and
+promote its widespread use.
+
+To contribute to this module, please visit http://odoo-community.org.
```
Comment (elicoidal) [inline stock_change_qty_reason/README.rst]: https

**Comment by elicoidal** [comment]
small details
:+1:

File: stock_change_qty_reason/models/stock.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+        move_id = super(StockInventoryLine,
+                        self)._resolve_inventory_line(inventory_line)
+
+        reason = self.env.context.get('change_quantity_reason', False)
+        if reason and move_id:
+            move = self.env['stock.move'].browse(move_id)
+
+            if move.origin:
+                move.origin = ' ,'.join(move.origin, reason)
```
Comment (sbidoul) [inline stock_change_qty_reason/models/stock.py]: This should be `move.origin = ', '.join([move.origin, reason])`

**Comment by Cedric-Pigeon** [comment]
:+1:

File: stock_change_qty_reason/README.rst (hunk 1 above)
Comment (elicoidal) [inline stock_change_qty_reason/README.rst:4]: Please update the document title format (add a line before the title)

**Comment by andhit-r** [comment]
Functionality tested on runbot :+1:

**Comment by Daniel-CA** [comment]
:+1: LGTM

**Pull request merged**